From 97fc639722f38aa78097e0db11243cab44d8e58a Mon Sep 17 00:00:00 2001
From: Fred Fan <r01011@freescale.com>
Date: Wed, 23 Apr 2008 10:09:12 +0800
Subject: [PATCH] ENGR00073908 Fix the wrong event mapping in pseudo irq on i.MX35 3stack

Fix the error shift value to get the mask of event register2 in MCU.
And change the mask argument to 0xFF when calling pmic_write_reg.

Signed-off-by: Fred Fan <r01011@freescale.com>
---
 arch/arm/mach-mx35/mx35_3stack_irq.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx35/mx35_3stack_irq.c b/arch/arm/mach-mx35/mx35_3stack_irq.c
index ba608a7..e7beabf 100644
--- a/arch/arm/mach-mx35/mx35_3stack_irq.c
+++ b/arch/arm/mach-mx35/mx35_3stack_irq.c
@@ -233,11 +233,11 @@ static void mcu_state_handler(struct work_struct *work)
 	int err, i;
 	unsigned int event1, event2;
 	event1 = pseudo_irq_enable & ((1 << MCU_INT_KEYPAD) - 1);
-	event2 = pseudo_irq_enable >> (MCU_INT_KEYPAD - 1);
+	event2 = pseudo_irq_enable >> MCU_INT_KEYPAD;
 
 	for (i = 0; i < 3; i++) {
-		err = pmic_write_reg(REG_MCU_INT_ENABLE_1, event1, event1);
-		err |= pmic_write_reg(REG_MCU_INT_ENABLE_2, event2, event2);
+		err = pmic_write_reg(REG_MCU_INT_ENABLE_1, event1, 0xFF);
+		err |= pmic_write_reg(REG_MCU_INT_ENABLE_2, event2, 0xFF);
 		if (err == PMIC_SUCCESS)
 			break;
 	}
-- 
1.5.4.4

